internal/brokers/broker.go

```diff
@@ -168,63 +177,54 @@ func (b Broker) cancelIsAuthorized(ctx context.Context, sessionID string) {
+	for _, layout := range supportedUILayouts {
+		layoutValidator := make(map[string]fieldValidator)
+		for key, value := range layout {
+			if key == "type" {
+				continue
+			}
 
-	typ := layout["type"]
-	label := layout["label"]
-	entry := layout["entry"]
-	button := layout["button"]
-	wait := layout["wait"]
-	content := layout["content"]
-
-	r = make(map[string]string)
-	switch typ {
-	case "form":
-		if label == "" {
-			return nil, fmt.Errorf("'label' is required")
-		}
-		if !slices.Contains([]string{"chars", "digits", "chars_password", "digits_password", ""}, entry) {
-			return nil, fmt.Errorf("'entry' does not match allowed values for this type: %v", entry)
-		}
-		if !slices.Contains([]string{"true", "false", ""}, wait) {
-			return nil, fmt.Errorf("'wait' does not match allowed values for this type: %v", wait)
-		}
-		r["label"] = label
-		r["entry"] = entry
-		r["button"] = button
-		r["wait"] = wait
-	case "qrcode":
-		if content == "" {
-			return nil, fmt.Errorf("'content' is required")
-		}
-		if !slices.Contains([]string{"true", "false"}, wait) {
-			return nil, fmt.Errorf("'wait' is required and does not match allowed values for this type: %v", wait)
-		}
-		r["content"] = content
-		r["wait"] = wait
-		r["label"] = label
-		r["entry"] = entry
-		r["button"] = button
-	case "newpassword":
-		if label == "" {
-			return nil, fmt.Errorf("'label' is required")
-		}
-		if !slices.Contains([]string{"chars", "digits", "chars_password", "digits_password"}, entry) {
-			return nil, fmt.Errorf("'entry' does not match allowed values for this type: %v", entry)
+			modifier, supportedValues, _ := strings.Cut(value, ":")
+			validator := fieldValidator{
+				supportedValues: nil,
+				required:        (modifier == "required"),
+			}
+			if supportedValues != "" {
+				validator.supportedValues = strings.Split(supportedValues, ",")
+			}
+			layoutValidator[key] = validator
 		}
-		r["label"] = label
-		r["entry"] = entry
-		r["button"] = button
-	case "webview":
-	default:
-		return nil, fmt.Errorf("invalid layout option: type is required, got: %v", layout)
+		validators[layout["type"]] = layoutValidator
 	}
+	b.layoutValidators = validators
+}
 
-	r["type"] = typ
+func (b Broker) validateUILayout(layout map[string]string) (r map[string]string, err error) {
+	defer decorate.OnError(&err, "could not validate UI layout")
+
+	typ := layout["type"]
+	layoutValidator, exists := b.layoutValidators[typ]
+	if !exists {
+		return nil, fmt.Errorf("no validator for UI layout type %q", typ)
+	}
 
+	r = map[string]string{"type": typ}
+	for key, validator := range layoutValidator {
+		value, exists := layout[key]
+		if !exists {
+			if validator.required {
+				return nil, fmt.Errorf("required field %q not found in UI layout", key)
+			}
+			continue
+		}
+
+		if validator.supportedValues != nil && !slices.Contains(validator.supportedValues, value) {
+			return nil, fmt.Errorf("field %q has invalid value type %q", key, value)
+		}
+		r[key] = value
```

Comment:
Ok, so this was a design decision that I'm not totally 100% sure of, but I thought it was worth pushing it so we can talk about it...

The thing is: we used to always assign everything to the returned layout even if it was an empty value (see the changes in the golden files). I'm not sure that was the best approach, tbh. 
I think it's best to validate and return only what was supposed to be there (i.e. required files or optional fields that were set) instead of returning everything. This means it will be up to the UI renderer to check if the values are there or not before rendering them and I think this is the best approach, since we end up preventing content injections that could come from the brokers.

wdyt?